tensorflow/python/ops/check_ops.py

```diff
@@ -1126,7 +1127,7 @@ def assert_rank(x, rank, data=None, summarize=None, message=None, name=None):
     ValueError:  If static checks determine `x` has wrong rank.
   """
   with ops.name_scope(name, 'assert_rank', (x, rank) + tuple(data or [])):
-    x = ops.convert_to_tensor(x, name='x')
+    x = x if sparse_tensor.is_sparse(x) else ops.convert_to_tensor(x, name='x')
```

Comment:
So this is very interesting. Why do we need the ops.convert_to_tensor at all? If it's just for `name` that seems bad. I would suggest simply removing `name` altogether here since it is empty with eager execution anyways. Then I don't believe we need to call `ops.convert_to_tensor` since we expect `x` to be a tensor.

Reply:
I suspect when the input is the type of **numpy array**,  **ops.convert_to_tensor** will work to change the type to a tensor.
The name is needed for this piece of code: https://github.com/tensorflow/tensorflow/blob/a1e5034c8ccb8f103c94f03f85cb4eaafda6e680/tensorflow/python/ops/check_ops.py#L1727-L1730
Which used to give the detail information when an error raises.

Reply:
Are numpy arrays supported here? If so, it seems like the correct fix is to only call `ops.convert_to_tensor` with numpy arrays.

As mentioned earlier, let's just drop the name from the error message as it seems unnecessary.

Reply:
Thanks for the input.
> Are numpy arrays supported here? If so, it seems like the correct fix is to only call ops.convert_to_tensor with numpy arrays.
Yes, numpy arrays is supported here. However, If I change to only call ops.convert_to_tensor with numpy arrays, the unit tests failed. After dig out, I find not only numpy arrays but also python list , constant_op.constant and so on may also be supported in the origin logic which would be converted to tensor at very beginning. So here I am afraid we had better pick sparse tensor out as a special input case.

> As mentioned earlier, let's just drop the name from the error message as it seems unnecessary.
There are lots of place which invokes tensor.name directly, we also need to drop these code... I am afraid that's too many. And I think the tensor name appears in the error message may be helpful to debug... And the change is less if we add the name property in the sparse tensor.
Thanks again for the review :)

Reply:
Thanks for the input.
> Are numpy arrays supported here? If so, it seems like the correct fix is to only call ops.convert_to_tensor with numpy arrays. 

Yes, numpy arrays is supported here. However, If I change to only call ops.convert_to_tensor with numpy arrays, the unit tests failed. After dig out, I find not only numpy arrays but also python list , constant_op.constant and so on may also be supported in the origin logic which would be converted to tensor at very beginning. So here I am afraid we had better pick sparse tensor out as a special input case.

> As mentioned earlier, let's just drop the name from the error message as it seems unnecessary.

There are lots of place which invokes tensor.name directly, we also need to drop these code... I am afraid that's too many. And I think the tensor name appears in the error message may be helpful to debug... And the change is less if we add the name property in the sparse tensor.
Thanks again for the review :)